src/HealthGPS/energy_balance_model.cpp

```diff
@@ -78,131 +82,169 @@ void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
 
-    // Update body fat and lean tissue.
-    double F = steady_F + (F_0 - steady_F) * exp(-365.0 / tau);
-    double L = steady_L + (L_0 - steady_L) * exp(-365.0 / tau);
+    // Compute body fat and lean tissue.
+    double F = steady_F - (steady_F - F_0) * exp(-365.0 / tau);
+    double L = steady_L - (steady_L - L_0) * exp(-365.0 / tau);
+
+    // Compute baseline adjustment coefficient.
+    double adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
 
-    // Update body weight.
+    // Compute body weight.
     double BW = F + L + G + W + ECF;
 
+    // Compute energy expenditure.
     double delta_BW = delta_0 * BW;
+    double EE = (shift + K + gamma_F * F + gamma_L * L + delta_BW + TEF + AT + EI * x) / (1.0 + x);
+
+    // New simulated person state.
+    SimulatePersonState state{.H = H,
+                              .BW = BW,
+                              .PAL = PAL,
+                              .RMR = RMR,
+                              .F = F,
+                              .L = L,
+                              .ECF = ECF,
+                              .G = G,
+                              .W = W,
+                              .EE = EE,
+                              .EI = EI,
+                              .adjust = adjust};
+
+    return state;
+}
+
+std::unordered_map<core::Identifier, double>
+EnergyBalanceModel::compute_nutrients(const Person &person) const {
+    std::unordered_map<core::Identifier, double> nutrient_intakes;
+
+    for (const auto &equation : nutrient_equations_) {
+        double food_intake = person.get_risk_factor_value(equation.first);
+
+        for (const auto &coefficient : equation.second) {
+            double delta_nutrient = food_intake * coefficient.second;
+            nutrient_intakes[coefficient.first] += delta_nutrient;
+        }
+    }
+
+    return nutrient_intakes;
+}
+
+double EnergyBalanceModel::compute_EI(
+    const std::unordered_map<core::Identifier, double> &nutrient_intakes) const {
+    double EI = 0.0;
 
-    // Update energy expenditure.
-    double EE =
-        (final_shift + K + gamma_F * F + gamma_L * L + delta_BW + TEF + AT + EI * x) / (1.0 + x);
-
-    if (final_run) { // Final run:
-        // Save state and return nothing.
-        person.risk_factors[H_key] = H;
-        person.risk_factors[BW_key] = BW;
-        person.risk_factors[PAL_key] = PAL;
-        person.risk_factors[F_key] = F;
-        person.risk_factors[L_key] = L;
-        person.risk_factors[ECF_key] = ECF;
-        person.risk_factors[EI_key] = EI;
-        person.risk_factors[EE_key] = EE;
-        return std::array<double, 2>{};
-    } else { // Trial run:
-        // return BW and baseline adjustment coefficient.
-        double trial_adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
-        return std::array<double, 2>{BW, trial_adjust};
+    for (const auto &coefficient : energy_equation_) {
```

Comment:
Again, maybe use a structured binding here to make it more obvious what `coefficient.first` and `coefficient.second` refer to.